src/GovUk.Education.ExploreEducationStatistics.Common/Cache/CacheAttribute.cs

```diff
@@ -0,0 +1,73 @@
+#nullable enable
+using System;
+using System.Threading.Tasks;
+using AspectInjector.Broker;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Cache
+{
+    /// <summary>
+    /// Base caching attribute that should be extended with specific
+    /// caching implementations e.g. blob storage, memory, Redis, etc.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    [Injection(typeof(CacheAspect), Inherited = true)]
+    public abstract class CacheAttribute : Attribute
+    {
+        /// <summary>
+        /// The base type of the cache key to use.
+        /// The <see cref="Key"/> must be assignable to this.
+        /// </summary>
+        protected virtual Type BaseKey => typeof(ICacheKey);
+
+        /// <summary>
+        /// Cache with highest priority is checked first.
+        /// 0 - Lowest, 255 - Highest.
+        /// </summary>
+        public byte Priority { get; init; } = 0;
```

Comment:
Cool so you would have more than one cache annotations, each with their own priority, and it would go through them in priority order looking for the first of the cache types to return a result? If none have a cache hit which one is set?

Reply:
Yep indeed :smile: idea would be to potentially have multiple levels of cache if required (kinda doubt it, but the option's there).

When there is no cache hit, we set the real result in all of the caches!